socorro/processor/rules/mozilla.py

```diff
@@ -650,9 +655,16 @@ def format_module(self, item):
         version = version.replace("/", "\\/")
 
         debugid = item.get("debug_id", self.NULL_DEBUG_ID)
-        debugid = self.BAD_DEBUGID_CHARACTERS.sub("", debugid)
+        debugid = self.BAD_HEXID_CHARACTERS.sub("", debugid)
+
+        codeid = item.get("code_id", None)
+        if codeid:
+            codeid = self.BAD_HEXID_CHARACTERS.sub("", codeid)
```

Comment:
kinda feel like we should be redacting whole invalid values, rather than just invalid characters, e.g.:
```suggestion
        if self.BAD_HEXID_CHARACTERS.match(debugid):
            debugid = ""

        codeid = item.get("code_id", None)
        if self.BAD_HEXID_CHARACTERS.match(codeid):
            codeid = ""
```
or at least just for codeid, since it can be omitted:
```suggestion
        debugid = self.BAD_HEXID_CHARACTERS.sub("", debugid)

        codeid = item.get("code_id", None)
        if self.BAD_HEXID_CHARACTERS.match(codeid):
            codeid = ""
```

Reply:
Crash ingestion is fun because we often get bit flips and bad encoding changes and compiler bugs and all manner of curiosities that result in slightly bad values. For things like this, I hedge towards "having some value that's indicative of what _should_ have been there" is better than "remove the value altogether because it's malformed". It's been helpful to have some signal to debug weird situations.

If it wasn't the wily world of crash ingestion, I'd go with your suggestion.

Reply:
This is definitely a difference between data platform ingestion where the end result is something we need to be clean and proper for data analysis and crash ingestion where the end result is something we need to give us signal we can use for debugging crashes.